libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,452 @@
+      countingStep *= 10;
+      this._roundShift++;
+    }
+  }
+  /** Holds the value of step internally */
+  private _step: number = 1;
+
+  /** Binding for the disabled state. */
+  @Input()
+  get disabled(): boolean {
+    return this._isDisabled;
+  }
+  set disabled(disabled: boolean) {
+    this._isDisabled = coerceBooleanProperty(disabled);
+    this._changeDetectionRef.markForCheck();
+  }
+  private _isDisabled: boolean = false;
+
+  /**
+   * The binding for the value of the slider.
+   */
+  @Input()
+  get value(): number {
+    return this._value;
+  }
+  set value(value: number) {
+    this._updateValue(coerceNumberProperty(value));
+  }
+  private _value: number = 0;
+
+  /** Updates the value if the update is triggered by the consumer. */
+  private _updateValue(value: number, userTriggered: boolean = true): void {
+    this._value = value;
+    // We only need to update if the update is coming from outside the component.
+    if (userTriggered) {
+      this._value$.next(roundToSnap(value, this.step, this._min, this._max));
+    }
+  }
+
+  /** Observer that completes on ngOnDestroy */
+  private _destroy$ = new Subject<void>();
+
+  /**
+   * Convert input string value to number and call
+   * roundToSnap takes care of snapping the values to the steps
+   */
+  inputValueChanged(event: Event): void {
+    this._inputFieldValue$.next(
+      +(event.currentTarget as HTMLInputElement).value,
+    );
+  }
+
+  /** Provides event for value change */
+  @Output() change = new EventEmitter<number>();
+
+  /** @internal Holds the track wrapper */
+  @ViewChild('trackWrapper', { static: true })
+  _trackWrapper: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the thumb */
+  @ViewChild('thumb', { static: true })
+  _thumb: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider fill */
+  @ViewChild('sliderFill', { static: true })
+  _sliderFill: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider background */
+  @ViewChild('sliderBackground', { static: true })
+  _sliderBackground: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider input field */
+  @ViewChild(DtInput, { static: true })
+  _sliderInput: DtInput;
+
+  constructor(
+    private _changeDetectionRef: ChangeDetectorRef,
+    private _zone: NgZone,
+    private _platform: Platform,
+  ) {}
+
+  ngOnInit(): void {
+    const initValue = roundToSnap(this.value, this.step, this._min, this._max);
+    if (this._value$.value !== initValue) {
+      this._updateValue(initValue);
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (this._platform.isBrowser && 'ResizeObserver' in window) {
+      this._observer = new window.ResizeObserver(() => {
+        this._resizeObserver$.next();
+      });
+
+      this._observer.observe(this._trackWrapper.nativeElement);
+    }
+
+    this._clientRect$ = merge(
+      this._resizeObserver$.pipe(debounceTime(50)),
+      of(null), // at least one initial trigger is needed
```

Comment:
you could use for that a `startWith(null)` operator then you don't have to merge it.

Reply:
I have tried make use of `startsWith`, but I struggled to achieve the same result. What would be the source observer that 'starts with' null?

Reply:
In a call, we agreed to leave this as is for now.